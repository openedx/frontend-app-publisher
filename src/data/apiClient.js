import { getAuthenticatedAPIClient } from '@edx/frontend-auth';

const apiClient = getAuthenticatedAPIClient({
  appBaseUrl: process.env.BASE_URL,
  authBaseUrl: process.env.LMS_BASE_URL,
  loginUrl: process.env.LOGIN_URL,
  logoutUrl: process.env.LOGOUT_URL,
  csrfTokenApiPath: process.env.CSRF_TOKEN_API_PATH,
  refreshAccessTokenEndpoint: process.env.REFRESH_ACCESS_TOKEN_ENDPOINT,
  accessTokenCookieName: process.env.ACCESS_TOKEN_COOKIE_NAME,
  userInfoCookieName: process.env.USER_INFO_COOKIE_NAME,
});

export default apiClient;
