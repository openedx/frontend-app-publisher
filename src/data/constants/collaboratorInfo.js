export const CREATE_COLLABORATOR = 'CREATE_COLLABORATOR';
export const CREATE_COLLABORATOR_SUCCESS = 'CREATE_COLLABORATOR_SUCCESS';
export const CREATE_COLLABORATOR_FAIL = 'CREATE_COLLABORATOR_FAIL';
export const EDIT_COLLABORATOR_INFO = 'EDIT_COLLABORATOR_INFO';
export const EDIT_COLLABORATOR_INFO_SUCCESS = 'EDIT_COLLABORATOR_INFO_SUCCESS';
export const EDIT_COLLABORATOR_INFO_FAIL = 'EDIT_COLLABORATOR_INFO_FAIL';
export const CANCEL_COLLABORATOR_INFO = 'CANCEL_COLLABORATOR_INFO';
export const RESET_COLLABORATOR_INFO = 'RESET_COLLABORATOR_INFO';
