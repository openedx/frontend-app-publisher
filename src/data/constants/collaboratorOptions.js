export const REQUEST_COLLABORATOR_OPTIONS = 'REQUEST_COLLABORATOR_OPTIONS';
export const REQUEST_COLLABORATOR_OPTIONS_FAIL = 'REQUEST_COLLABORATOR_OPTIONS_FAIL';
export const REQUEST_COLLABORATOR_OPTIONS_SUCCESS = 'REQUEST_COLLABORATOR_OPTIONS_SUCCESS';
