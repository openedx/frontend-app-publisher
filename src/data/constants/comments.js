const CREATE_COMMENT = 'CREATE_COMMENT';
const CREATE_COMMENT_FAIL = 'CREATE_COMMENT_FAIL';
const CREATE_COMMENT_SUCCESS = 'CREATE_COMMENT_SUCCESS';
const REQUEST_COMMENTS = 'REQUEST_COMMENTS';
const REQUEST_COMMENTS_FAIL = 'REQUEST_COMMENTS_FAIL';
const REQUEST_COMMENTS_SUCCESS = 'REQUEST_COMMENTS_SUCCESS';

export {
  CREATE_COMMENT,
  CREATE_COMMENT_FAIL,
  CREATE_COMMENT_SUCCESS,
  REQUEST_COMMENTS,
  REQUEST_COMMENTS_FAIL,
  REQUEST_COMMENTS_SUCCESS,
};
