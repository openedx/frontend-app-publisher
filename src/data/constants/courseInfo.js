const REQUEST_COURSE_INFO_FAIL = 'REQUEST_COURSE_INFO_FAIL';
const REQUEST_COURSE_INFO_SUCCESS = 'REQUEST_COURSE_INFO_SUCCESS';
const REQUEST_COURSE_INFO = 'REQUEST_COURSE_INFO';
const CREATE_COURSE = 'CREATE_COURSE';
const CREATE_COURSE_SUCCESS = 'CREATE_COURSE_SUCCESS';
const CREATE_COURSE_FAIL = 'CREATE_COURSE_FAIL';
const EDIT_COURSE_INFO = 'EDIT_COURSE_INFO';
const EDIT_COURSE_SUCCESS = 'EDIT_COURSE_SUCCESS';
const EDIT_COURSE_FAIL = 'EDIT_COURSE_FAIL';
const CREATE_COURSE_RUN = 'CREATE_COURSE_RUN';
const CREATE_COURSE_RUN_SUCCESS = 'CREATE_COURSE_RUN_SUCCESS';
const CREATE_COURSE_RUN_FAIL = 'CREATE_COURSE_RUN_FAIL';

const UUID_REGEX = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/;

export {
  REQUEST_COURSE_INFO_FAIL,
  REQUEST_COURSE_INFO_SUCCESS,
  REQUEST_COURSE_INFO,
  CREATE_COURSE,
  CREATE_COURSE_SUCCESS,
  CREATE_COURSE_FAIL,
  EDIT_COURSE_INFO,
  EDIT_COURSE_SUCCESS,
  EDIT_COURSE_FAIL,
  CREATE_COURSE_RUN,
  CREATE_COURSE_RUN_SUCCESS,
  CREATE_COURSE_RUN_FAIL,
  UUID_REGEX,
};
