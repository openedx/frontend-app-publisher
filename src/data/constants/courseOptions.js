const REQUEST_COURSE_OPTIONS_FAIL = 'REQUEST_COURSE_OPTIONS_FAIL';
const REQUEST_COURSE_OPTIONS_SUCCESS = 'REQUEST_COURSE_OPTIONS_SUCCESS';
const REQUEST_COURSE_OPTIONS = 'REQUEST_COURSE_OPTIONS';

export {
  REQUEST_COURSE_OPTIONS_FAIL,
  REQUEST_COURSE_OPTIONS_SUCCESS,
  REQUEST_COURSE_OPTIONS,
};
