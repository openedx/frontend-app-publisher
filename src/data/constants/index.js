const VERIFIED_TRACK = Object.freeze({ name: 'Verified and Audit', key: 'verified' });
const AUDIT_TRACK = Object.freeze({ name: 'Audit Only', key: 'audit' });
const PROFESSIONAL_TRACK = Object.freeze({ name: 'Professional Only', key: 'professional' });
const MASTERS_TRACK = Object.freeze({ key: 'masters' });
const ENTITLEMENT_TRACKS = Object.freeze(['verified', 'professional']);

const REVIEW_BY_LEGAL = 'review_by_legal';
const REVIEW_BY_INTERNAL = 'review_by_internal';
const PUBLISHED = 'published';
const REVIEWED = 'reviewed';
const UNPUBLISHED = 'unpublished';
const ARCHIVED = 'archived';
const IN_REVIEW_STATUS = Object.freeze([REVIEW_BY_LEGAL, REVIEW_BY_INTERNAL]);
const POST_REVIEW_STATUSES = Object.freeze([PUBLISHED, REVIEWED]);
const COURSE_EXEMPT_FIELDS = Object.freeze(['imgSrc', 'videoSrc']);
const COURSE_RUN_NON_EXEMPT_FIELDS = Object.freeze(['expected_program_name', 'expected_program_type']);

const DATE_FORMAT = 'YYYY-MM-DD[T]HH:mm:ss[Z]';
const DATE_INPUT_PATTERN = '20[1-9][0-9]/(0[1-9]|1[012])/(0[1-9]|[12][0-9]|3[01])';
const FORMAT_DATE_MATCHER = /20\d{2}-(0\d{1}|1[0-2])-([0-2]\d{1}|3[0-1])/;
const NORMALIZE_DATE_MATCHER = /20\d{2}\/(0\d{1}|1[0-2])\/([0-2]\d{1}|3[0-1])/;

const EXECUTIVE_EDUCATION_SLUG = 'executive-education-2u';
const BOOTCAMP_SLUG = 'bootcamp-2u';

const COURSE_URL_SLUG_PATTERN_OLD = '^[a-z0-9_]+(?:-[a-z0-9_]+)*$';
const COURSE_URL_SLUG_PATTERN_NEW = '^learn/[a-z0-9_]+(?:-?[a-z0-9_]+)*/[a-z0-9_]+(?:-?[a-z0-9_]+)*$';
const COURSE_URL_SLUG_PATTERN = `${COURSE_URL_SLUG_PATTERN_NEW}|${COURSE_URL_SLUG_PATTERN_OLD}`;

const COURSE_URL_SLUG_VALIDATION_MESSAGE = {
  [COURSE_URL_SLUG_PATTERN_OLD]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only.',
  [COURSE_URL_SLUG_PATTERN_NEW]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only and must be in the format learn/<primary_subject>/<org-slug>-<course_slug>',
  [COURSE_URL_SLUG_PATTERN]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only and must be in the format <custom-url-slug> or learn/<primary_subject>/<org-slug>-<course_slug>.',
};

export {
  VERIFIED_TRACK,
  AUDIT_TRACK,
  PROFESSIONAL_TRACK,
  MASTERS_TRACK,
  ENTITLEMENT_TRACKS,
  REVIEW_BY_LEGAL,
  REVIEW_BY_INTERNAL,
  PUBLISHED,
  REVIEWED,
  UNPUBLISHED,
  ARCHIVED,
  IN_REVIEW_STATUS,
  POST_REVIEW_STATUSES,
  DATE_FORMAT,
  DATE_INPUT_PATTERN,
  FORMAT_DATE_MATCHER,
  NORMALIZE_DATE_MATCHER,
  COURSE_EXEMPT_FIELDS,
  COURSE_RUN_NON_EXEMPT_FIELDS,
  EXECUTIVE_EDUCATION_SLUG,
  BOOTCAMP_SLUG,
  COURSE_URL_SLUG_PATTERN,
  COURSE_URL_SLUG_PATTERN_NEW,
  COURSE_URL_SLUG_PATTERN_OLD,
  COURSE_URL_SLUG_VALIDATION_MESSAGE,
};
