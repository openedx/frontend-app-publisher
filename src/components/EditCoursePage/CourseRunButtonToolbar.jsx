import PropTypes from 'prop-types';
import React from 'react';

import ButtonToolbar from '../ButtonToolbar';
import CourseRunSubmitButton from './CourseRunSubmitButton';

import { PUBLISHED, UNPUBLISHED } from '../../data/constants';

function CourseRunButtonToolbar(props) {
  const {
    className,
    disabled,
    editable,
    hasNonExemptChanges,
    onSubmit,
    pristine,
    status,
    submitting,
  } = props;

  if (!editable) {
    return null;
  }

  if (status === PUBLISHED) {
    if (pristine) {
      return null;
    }
    return (
      <div className="font-italic text-center">
        To publish changes, click ‘Save & Re-Publish’ below.
      </div>
    );
  }

  return (
    <ButtonToolbar className={className}>
      <CourseRunSubmitButton
        disabled={disabled}
        hasNonExemptChanges={hasNonExemptChanges}
        onSubmit={onSubmit}
        status={status}
        submitting={submitting}
      />
    </ButtonToolbar>
  );
}

CourseRunButtonToolbar.propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.bool,
  editable: PropTypes.bool,
  hasNonExemptChanges: PropTypes.bool,
  onSubmit: PropTypes.func,
  pristine: PropTypes.bool,
  status: PropTypes.string,
  submitting: PropTypes.bool,
};

CourseRunButtonToolbar.defaultProps = {
  className: '',
  disabled: false,
  editable: false,
  hasNonExemptChanges: false,
  onSubmit: () => {},
  pristine: false,
  status: UNPUBLISHED,
  submitting: false,
};

export default CourseRunButtonToolbar;
