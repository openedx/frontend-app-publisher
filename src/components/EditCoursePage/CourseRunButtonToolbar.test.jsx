import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CourseRunButtonToolbar from './CourseRunButtonToolbar';
import { PUBLISHED } from '../../data/constants';

describe('Course Run Button Toolbar', () => {
  it('default parameters', () => {
    const component = shallow(<CourseRunButtonToolbar />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('editable', () => {
    const component = shallow(<CourseRunButtonToolbar editable />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('published pristine', () => {
    const component = shallow(<CourseRunButtonToolbar editable pristine status={PUBLISHED} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('published with changes', () => {
    const component = shallow(<CourseRunButtonToolbar editable status={PUBLISHED} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
