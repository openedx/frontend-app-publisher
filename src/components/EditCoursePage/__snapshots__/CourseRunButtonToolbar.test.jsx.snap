// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Run Button Toolbar default parameters 1`] = `""`;

exports[`Course Run Button Toolbar editable 1`] = `
<ButtonToolbar
  className=""
  leftJustify={false}
>
  <CourseRunSubmitButton
    disabled={false}
    hasNonExemptChanges={false}
    onSubmit={[Function]}
    status="unpublished"
    submitting={false}
  />
</ButtonToolbar>
`;

exports[`Course Run Button Toolbar published pristine 1`] = `""`;

exports[`Course Run Button Toolbar published with changes 1`] = `
<div
  className="font-italic text-center"
>
  To publish changes, click ‘Save & Re-Publish’ below.
</div>
`;
